zng/docs/spec.md

```diff
@@ -547,9 +575,9 @@ grammar describing the textual type encodings is:
 A reference implementation of this type system is embedded in
```

Comment:
This comment seems a little out of place just after the TZNG pseudo-grammar. Maybe it should go at the end of section 3?

Reply:
@henridf: Good catch. I agree it seems out of place. I'm not sure if it belongs at the end of section 3 either. Here's some thinking out loud...

I looked back at the history of this. As I suspected, the line has been there since [the first version of the ZSON spec](https://github.com/brimsec/zq/blob/5fc47f71c7f9b130c3bcd4a2d377735349fd39a9/pkg/zson/docs/spec.md) when there was no binary format yet and it was trying to be backward compatible with Zeek, and hence was pointing at what was then `zq/pkg/zeek` as the reference implementation, since at the time we were implementing the Zeek data types in their entirety. It seems as if the line has lived on and morphed we moved packages around and cut the ties on 100% backward compatibility with Zeek.

Now, even without me fully understanding the code, no doubt that TZNG psuedo-grammar is implemented _somewhere_ in zq. What I can't say is precisely where, or if it's spread out, if the line is helpful as currently written, as it seems to hand-wave at a point in the code tree that implements lots of other things too.

Up-leveling a bit, wouldn't it be fair to say that zq acts as a reference implementation for _everything_ in the ZNG spec? That is, not just the implementation of the type systems in both TZNG and binary ZNG, but also the reader behaviors, error handling, etc. The opening disclaimer at the top of the spec effectively already says that zq is this reference implementation. It seems one thing we could do is remove the line from this spot below the TZNG pseudo-grammar, make it more generic, and move it to the bottom of the Introduction. After all, it seems a fitting way to close that paragraph would be to remind the reader that the spec is backed up by running code.

... or... if we think it's helpful to the reader to give _precise_ pointers to places within the zq source code to where certain parts of the spec are implemented, we could do that, starting by perhaps refining this one to point specifically to where the TZNG grammar is implemented, then for consistency, adding the same at other places in the spec. This feels like a lot of work that I'd need to lean on you Dev guys for, though. Considering it remains to be seen how many people are scrutinizing the spec at that level of detail, I'm _not_ personally advocating that we make that investment. If people start coming to us with such questions on Slack we could answer them a la carte until it becomes such a nuisance that we add this to the spec.

I'm open to other suggestions as well.

Reply:
I agree with making the sentence more generic and moving it to the intro. And also that doing detailed references to specific parts of the implementation isn't worthwhile for the time being.

Reply:
@henridf: I've made it generic and moved it to the bottom of the intro. PTAL and let me know what you think. Thanks!

Reply:
👍

Reply:
👍